File: src/main/go/leetcode/lists/hot100/LeetCode128.go
package hot100

func longestConsecutive(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	set := make(map[int]bool)
	for _, nu := range nums {
		set[nu] = true
	}
	var res int
	for nu := range set {
		temp := 1
		if _, ok := set[nu-1]; !ok {
			for set[nu+1] {
				temp++
				nu++
			}
		}
		res = Max(res, temp)
	}
	return res
}

File: src/main/java/leetcode/lists/hot100/LeetCode128.java
package leetcode.lists.hot100;

import java.util.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/15 23:35
 */
public class LeetCode128 {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        HashSet<Integer> set = new HashSet<>();
        for (int n : nums) {
            set.add(n);
        }
        int max = 0;

        for (int j : nums) {
            int num = j;
            int temp = 1;
            if (!set.contains(num - 1)) {
                while (set.contains(num + 1)) {
                    num++;
                    temp++;
                }
                max = Math.max(temp, max);
            }
        }
        return max;
    }
}
